src/Data/ByteString/Lazy/Base16.hs

```diff
@@ -171,5 +171,7 @@ isBase16 bs = isValidBase16 bs && isRight (decodeBase16Untyped bs)
 -- True
 --
 isValidBase16 :: ByteString -> Bool
-isValidBase16 = all (`elem` "0123456789abcdefABCDEF")
+isValidBase16 bs = case bs of
+    Empty -> True
+    Chunk sbs next -> B16.isValidBase16 sbs && isValidBase16 next
```

Comment:
this is equivalent to a [foldrChunks](https://hackage.haskell.org/package/bytestring-0.12.0.2/docs/src/Data.ByteString.Lazy.Internal.html#foldrChunks) for lazy bytestrings.